sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -895,8 +897,9 @@ ur_program_handle_t ProgramManager::getBuiltURProgram(
   std::copy(DeviceImagesToLink.begin(), DeviceImagesToLink.end(),
             std::back_inserter(AllImages));
 
-  return getBuiltURProgram(std::move(AllImages), ContextImpl,
-                           {std::move(Device)});
+  return getBuiltURProgram(
+      std::move(AllImages), ContextImpl,
+      {std::move(createSyclObjFromImpl<device>(RootOrSubDevImpl))});
```

Comment:
```suggestion
      {createSyclObjFromImpl<device>(RootOrSubDevImpl)});
```

return value is already an r-value.